### DISCUSSION TITLE
Potential Software Integration

### DISCUSSION THREAD

**Discussion by Jedidiah7797** [getredash/redash#5999] [Q&A]
Hello guys! Could I integrate Redash with our software? We (Cachengo) have created a cloud-based IT Marketplace for developers and want Redash to be part of it. Could this be possible if so, how can we make this possible?

**Comment by justinclift** [comment]
Are you asking legally, or how to do it technically, or something else? :smile:

**Reply by Jedidiah7797** [reply]
Well kinda both actually😅. I’m trying to get in touch with the right person to explain what I’m talking about.

**Reply by justinclift** [reply]
Probably do it here?

**Reply by Jedidiah7797** [reply]
Sure! Cachengo App Marketplace is a decentralized cloud-based platform where software developers or other users can process, clean, share, analyze, model, and monetize their datasets with solutions from BI to ML. Users can build and deploy data engineering workflows, machine learning models, analytics dashboards, and more. With this, you can build highly scalable apps and more. If Redash could be on this platform, how could that be possible?

**Reply by justinclift** [reply]
Are you thinking to have Redash as some kind of packaged offering in the "Marketplace", or are you more meaning to integrate it into the Marketplace platform itself so all of the users can visualise their data? :smile:

**Reply by Jedidiah7797** [reply]
Haha yes Justin! more like a packaged offering if that makes sense.

**Reply by justinclift** [reply]
Yeah, that makes sense.  Redash itself is working out a new maintainership / Community-led approach, which is just now kicking off.

So timing wise, this needs to wait probably a month or so for us to get that sorted out first + have an updated Redash release that's suitable to deploy.  The current one has some bugs that definitely need fixing first.

Are you ok to check back with us in a few weeks? :smile:

**Reply by Jedidiah7797** [reply]
Yes Justin, thanks. Here's my contact information 

Jedidiah.Ikechukwu@cachengo.com

**Reply by gylaz** [reply]
> to integrate it into the Marketplace platform itself so all of the users can visualise their data

I am curious, is it possible to do this today with Redash? Meaning, to integrate it into our app, to give users the feature-set of Redash. Maybe via an iFrame or somekind of React/JavaScript integration.

**Reply by justinclift** [reply]
@gylaz It depends on how much time and effort your developers can put into it.

It's not a straight up white-box solution (aka "add your logos here and here, then you're good to go"), though if your developers are reasonably ok with Python you could potentially turn it into one. :smile:

Redash already has the ability to provide publicly accessible charts, and dashboards (several charts on one page).  There is "technically" an API available that's mostly undocumented (!), which can do almost everything the official web interface does.  So - with some time and effort - you could hook up the Redash backend but provide your own front end if that'd be useful.

There's also a Go based SDK for the above mentioned API, which could be useful.  I've only played with that a small bit, but it seemed to be decent.

Does any of that help? :smile:

**Reply by gylaz** [reply]
> So - with some time and effort - you could hook up the Redash backend but provide your own front end if that'd be useful.

Yes, thank you. This sums it up nicely.

**Reply by justinclift** [reply]
Cool. :smile:
